02.wo_Defects/underrun_st.c

```diff
@@ -112,7 +112,7 @@ void underrun_st_008 ()
 	for (i = 0; i < 5; i ++)
 	{
 		*p = 1; /*Tool should not detect this line as error*/ /*No ERROR:Data Underrun*/
-		p --;
+                if (i < 4) p --;
```

Comment:
p-- will set p to a value before the beginning of the array if it executes on the final loop iteration. This is considered a buffer underrun even though the pointer is never dereferenced.